virtme/commands/run.py

```diff
@@ -1293,28 +1301,51 @@ def do_it() -> int:
     else:
         virtme_init_cmd = "virtme-init"
 
+    initsh = []
+    if args.systemd:
+        kernelargs.append("virtme_systemd=1")
+        kernelargs.append(f"console={arch.serial_console_args()[0]}")
+        initsh.extend(
+            ["mount --bind /dev/null " + unit for unit in get_conf("systemd.masks")]
+        )
+
     if args.root == "/":
-        initcmds = [f"init={guest_tools_path}/{virtme_init_cmd}"]
+        if args.systemd:
+            initsh.append(f"SYSTEMD_UNIT_PATH={CACHE_DIR}: exec /sbin/init")
+            initcmds = ["init=/bin/sh", "--", "-c", ";".join(initsh)]
+        else:
+            initcmds = [f"init={guest_tools_path}/{virtme_init_cmd}"]
     else:
         virtfs_config = VirtFSConfig(
             path=guest_tools_path,
             mount_tag="virtme.guesttools",
         )
         export_virtfs(qemu, arch, qemuargs, virtfs_config)
-        initcmds = [
-            "init=/bin/sh",
-            "--",
-            "-c",
-            ";".join(
+        virtfs_config = VirtFSConfig(
+            path=str(CACHE_DIR),
+            mount_tag="virtme.cache",
+        )
+        export_virtfs(qemu, arch, qemuargs, virtfs_config)
+        initsh.extend(
+            [
+                "mount -t tmpfs run /run",
+                "mkdir -p /run/virtme/guesttools",
+                "/bin/mount -n -t 9p -o ro,version=9p2000.L,trans=virtio,access=any "
+                + "virtme.guesttools /run/virtme/guesttools",
+            ]
+        )
+        if args.systemd:
+            initsh.extend(
                 [
-                    "mount -t tmpfs run /run",
-                    "mkdir -p /run/virtme/guesttools",
+                    "mkdir -p /run/virtme/cache",
                     "/bin/mount -n -t 9p -o ro,version=9p2000.L,trans=virtio,access=any "
-                    + "virtme.guesttools /run/virtme/guesttools",
-                    f"exec /run/virtme/guesttools/{virtme_init_cmd}",
+                    + "virtme.cache /run/virtme/cache",
+                    "SYSTEMD_UNIT_PATH=/run/virtme/cache: exec /sbin/init",
                 ]
-            ),
-        ]
+            )
+        else:
+            initsh.append(f"exec /run/virtme/guesttools/{virtme_init_cmd}")
+        initcmds = ["init=/bin/sh", "--", "-c", ";".join(initsh)]
```

Comment:
```suggestion
        initcmds = ["init=/bin/sh", "--", "-c", "; ".join(initsh)]
```
Makes it easier to read... But a little out-of-scope... :)